_use_cases/schema_validation_and_enforecement.markdown

```diff
@@ -0,0 +1,95 @@
+---
+name: Schema Validation and Enforcement
+---
+
+### Schema Validation and Enforcement
+
+#### Why
+
+In Kafka, producing applications transfer messages to consuming applications.  In order for messages to be transferred successfully
+producer and consumer need to agree on the format of the message being transferred. If there is a mismatch between the format sent
+by the producer and the expectations of the consumer, problems will result.  The consuming application could take the wrong business
+action or the consuming application may fail leading to complete failure of the system.  This problem is known as a *poison message*
+scenario.
+
+| ![image]({{'/assets/pages/images/schema_validation_problem.png' | absolute_url}}){:width="100%"} |
+|:-----------------------------------------------------------------:|
+|    *Problem: Poison message leading to consumer crash*     |
+
+If there is only a small number of applications, perhaps all managed by the same team, it is possible for the developers to use
+informal agreements about what message format will be used on which topic.  However, as the use of Kafka grows within the organization,
+the amount of effort required to keep all the formats in agreement grows.  Mistakes, and system downtime, become inevitable.
+
+##### Schemas to the rescue
+
+To overcome the problem, the Kafka **client** ecosystem supports _schemas_.  Schemas provide a programmatic description
+of the message format.
+
+Producers use a schema to advertise the format of the messages they send.  The consumer uses the same schema to help it
+decode the incoming message with the certainty that it conforms to the expected format.
+
+The schemas themselves could be packaged as part of the application.  Whilst the approach is functional, it is quite
+inflexible as it means all applications need to be redeployed every time a schema changes.  This becomes a challenge
+as the number of applications in the system grows.  To counter this challenge organizations often choose to deploy a
+*Schema Registry*. 
+
+##### Schema Registry
+
+The role of the  **Schema Registry**  is to maintain a library of schemas that are in-use within an organisation.
+
+Producer applications call the registry to retrieve the schema for the messages they need to send.
+
+Consuming applications use information embedded in the incoming message to be able to retrieve the right schema from
+the registry.  This allows the consumer to decode the message.
+
+| ![image]({{'/assets/pages/images/schema_validation_schema_registry.png' | absolute_url}}){:width="100%"} |
+|:-----------------------------------------------------------------:|
+|    *Kafka applications making use of a Schema Registry*     |
+
+#### So, what's the problem?
```

Comment:
Maybe we should also point out that maybe not all clients are schema-aware, we may have technology that wants to emit JSON without awareness of how to encode apicurio identifiers into the message. The schema validation check can enforce the schema and prevent invalid messages from these clients entering the topic.

Reply:
might just muddy the usecase though

Reply:
I think we should go with it as it stands.  I'm sure we'll want to improve the use-case documentation as we grow the feature.